### ISSUE TITLE
Bugs with single contact / opening group message with a single contact boots me out/crashes app

### ISSUE THREAD

**Issue by FriendlessD** [signalapp/Signal-Android#10742]
- [x] I have searched open and closed issues for duplicates
- [x] I am submitting a bug report for existing functionality that does not work as intended
- [x] I have read https://github.com/signalapp/Signal-Android/wiki/Submitting-useful-bug-reports
- [x] This isn't a feature request or a discussion topic

----------------------------------------

### Bug description

This is probably a few issues but I am convinced they are symptoms of one issue because they are only happening in things related to a single contact, N.

Every time I go into one specific groupchat, very soon after I select the message typing window, the app goes white and boots me back to the main conversation list or completely crashed the app. All other groupchats I have work fine.

This specific groupchat also includes a contact, N, whose messages I cannot see in any groupchat, this one or a couple other test groupchats with me, N, and one other. My individual messages mostly work with him except there are multiple threads that seem to alternate depending on if a picture/gif was sent or not (e.g. message incoming chat 1, one of us send MMS and now messages coming in one chat 2). I also tried to fix by setting signal as default SMS app and the signal threads were still there under his name then a third new message thread under "unknown" was opened. 

I think there is an issue with his phone number and my signal. N uses Google Fi if that matters. I use ATT. We both have pixel 3a. I have no issues with other contacts that also have pixel 3a. Don't for sure know anyone else that has Google Fi.

### Steps to reproduce

**Group Chat Crash**
- open app
- Open group message
- press chat bubble to begin typing message or start typing a couple letters
**Actual result:** 
Either boots back to the main conversation list or crashed app
**Expected result:**
Let me type my message to the group

**Individual Contact Chat Shenanigans**

Double Threads
- Write message to N in thread 1
- converse in thread 1
- one of us sends a picture
- messages move to thread 2
- send another picture
- messages move back to thread 1
**Actual result:** 
two individual message threads for this one contact
**Expected result:**
One individual message thread for this one contact

Can't see specific contact "N" messages in any groupchat
- Contact "N" writes message
- do not see it show up
- another contact in the same groupchat replies to "N"
- I see other other contacts message reply like normal with reply message and N's message along with "couldn't find original message"
**Actual result:** 
Cannot receive group messages from contact "N"
**Expected result:**
receive group messages from contact "N"

Default SMS App
- set signal as default SMS app
- new thread created with SMS texts (sms app: google messages) from N labeled as "unknown"
- now there are 3 threads for N (N individual signal message thread 1, N individual signal thread 2, and "unknown" thread with SMS texts)
**Actual result:** 
signal does not properly import N's contact information and imports N's google message SMS messages into an "unknown thread". Does not connect with signal contact N which has same phone number
**Expected result:**
1 thread for 1 contact

### Screenshots

![signal-2021-01-19-204114_001](https://user-images.githubusercontent.com/77546056/105130000-14c7c800-5a9b-11eb-9c78-b14224c94500.png)
![signal-2021-01-19-204114_002](https://user-images.githubusercontent.com/77546056/105130011-18f3e580-5a9b-11eb-8da6-fd6dfb0f8c01.png)
![signal-2021-01-19-204114_003](https://user-images.githubusercontent.com/77546056/105130018-1abda900-5a9b-11eb-91d9-37508efee7e9.png)
![signal-2021-01-19-204114_005](https://user-images.githubusercontent.com/77546056/105130072-345ef080-5a9b-11eb-9317-3b54978f73f3.png)
![signal-2021-01-19-204114_006](https://user-images.githubusercontent.com/77546056/105130075-35901d80-5a9b-11eb-9611-8ea8d3112488.png)
![signal-2021-01-19-204114_007](https://user-images.githubusercontent.com/77546056/105130076-3628b400-5a9b-11eb-9bb0-b5e6741f0b4b.png)
![signal-2021-01-19-204114_008](https://user-images.githubusercontent.com/77546056/105130078-3628b400-5a9b-11eb-909a-40b88ea527d3.png)
![signal-2021-01-19-204114_004](https://user-images.githubusercontent.com/77546056/105130079-36c14a80-5a9b-11eb-89a3-c0baee67d672.png)

### Device info
**Device:** Google Pixel 3a
**Android version:** 11
**Signal version:** 5.2.3

### Link to debug log
https://debuglogs.org/d357927762ab431a2486c9270e310d05bb36ad8baa0df65b47b731967947d96e>

**Comment by Caerbannog** [comment]
Fixed link to debug log: https://debuglogs.org/d357927762ab431a2486c9270e310d05bb36ad8baa0df65b47b731967947d96e

Hi @FriendlessD !

I've looked at your debug log. Note: I'm not a member of the Signal team.
I contains two notable lines as far as I can tell.

> org.thoughtcrime.securesms.recipients.Recipient$MissingAddressError (repeated multiple times)
> java.lang.AssertionError: RecipientId::**166** - No UUID or phone number!

So your contact with ID **166** might be part of the problem. It would be useful to check if this is your "N" contact. Unfortunately I haven't found a way for you to to check this.

One of these errors is triggered by this condition:
```
if (!recipient.getUuid().isPresent() && !recipient.getE164().isPresent()) {
  throw new AssertionError(recipient.getId() + " - No UUID or phone number!");
}
```

An E164 phone number looks like this: +33607499999 (the +33 prefix depends on your country).
**Can you check if your "N" contact profile has such a phone number format in your address book, accessed from Signal? A screenshot and steps would be great.**

**Comment by bieriii** [comment]
Hello,

I experience the same bug.
In the case of the crashing group chat, I was able to type and send a group message as long as I was in flight mode.
For the contact which seems to be the issue, I checked, that the phone number is correct format in the contact list. The contact which seemed to be the issue appeared in the group members list as "unknown" as long as I was connected to the network. In flight mode it showed up as the correct contact.

### Device info

**Device:** Samsung A71
**Android version:** 10
**Signal version:** 5.2.3

### Link to debug log

https://debuglogs.org/53bc51736ee0fa484a83be42e4097ce99bea738419976171c606ca6240bea06b

**Comment by FriendlessD** [comment]
Hi @Caerbannog,

I checked and my phone contacts have my country code (+1) on them for the most part. N did not have a +1 but when I added it, there was no change. Other contacts do not have +1 and have been working just fine. Other members of the groupchat that crashes all the time have phone numbers in my contact list without the country codes and I have no issues with them.

@bieriii putting my phone in airplane mode worked just like yours! I also tried airplane mode + wifi and the issue returned so it must be something that happens when signal connects to the internet? Were you able to find a way to fix this issue so you don't have to have your phone in airplane mod?e

Thank you both!

**Comment by bieriii** [comment]
Hello @FriendlessD ,
I was able so that the problem doesn't happen anymore for now, but I did multiple things and don't know what was the solution. I left the group and reentered it. I deleted the phone number of the "bad" contact and removed all connections with all the apps, then I deleted the cache of signal, rebooted and added the phone number again to my contact. Somehow this fixed it, although when I check my contact in signal, there seems to be still a problem, because I am in multiple groups with this contact and if I click on the contact in the group members window, it shows always just the current group as a member for that contact. This is not the case for other contacts, where it shows all the groups they are also member of.

**Comment by loyhaim** [comment]
Hi, after few month with this bug, I think I have finally found a workaround.

I initially changed the phone numbers of problematic contacts like described above, removing the international numbers +33 but nothing changed.

I recently realized that even changing the name of the contact didn't change it in signal. Actually, I changed the contact on my Gmail account, but not on my android phone, even if both were linked.

When I removed the local problematic contacts on my phone, keeping only Gmail part , I started to receive their messages again. Wonderful!

Hope it helps someone!
